### DISCUSSION TITLE
Vector3 and Matrix4 Operation

### DISCUSSION THREAD

**Discussion by VirusPC** [galacean/discussion#114] [Engine] [answered]
In `three.js`, we can use `vector3.applyMatrix4` to do vector-matrix multiplication.
![image](https://github.com/galacean/discussion/assets/30697592/fe32a803-613c-463c-bcc2-654042a5213c)

In galacean, you can do the similar operation with `Vector3.transformToVec3`. However, it does not put `w` into consideration.
![image](https://github.com/galacean/discussion/assets/30697592/bc7df993-f8e9-42ae-90b6-ec61b55c54ca)

Since I know that multiplying Vector3 by Matrix4 is mathematically illegal, I wonder why `galacean` designed this way, why galacean behaves differently from `three.js`?

**Comment by cptbtptpbcptdtptp** [comment]
其实是顶点与向量之间的关系，这里引用一下闫老师的介绍（请在 00:58:30 左右观看）
https://www.bilibili.com/video/BV1X7411F744?p=3&vd_source=b6799713f5c41a29490a98ffe0a7157d

**Comment by GuoLei1990** [comment, marked as answer]
What you need is `transformCoordinate`

**Reply by VirusPC** [reply]
Thanks!
